doc improvement: Gazell restructure (#6482)
nrfconnect/sdk-nrf: samples/gazell/gzp_dynamic_pairing_host/README.rst

.. _gzp_dynamic_pairing_host:

Gazell Dynamic Pairing Host
###########################

This sample demonstrates the functionality of the :ref:`gzp` subsystem.
It consists of two applications, one running on the Device and one running on the Host.
You must follow the instructions and use of files from this page, and the :ref:`gzp_dynamic_pairing_device` sample.

.. contents::
   :local:
   :depth: 2

Requirements
************

.. note::

>    You must follow the steps from and include the :ref:`_gzp_dynamic_pairing_device` sample in addition to this sample.

Reviewer: ```suggestion
   Follow the steps from and include the :ref:`_gzp_dynamic_pairing_device` sample in addition to this sample.
```

* Device
   The application sends packets continuously.
   If a packet transmission fails (either times out or encryption fails), the Device makes an attempt to pair with a Host by sending a pairing request, consisting of an "address request" and a "Host ID" request.
   If the Device is paired with a Host, the pairing data is stored into the non-volatile memory.
   Before adding a packet to the TX queue, the content of the buttons is copied to the first payload byte (byte 0).
   The application alternates between sending the packets encrypted through the pairing library or directly as plaintext.

* Host
   The application listens for packets continuously, monitoring for pairing requests as well as normal user data.
   The Gazell pairing library uses pipe 0 and pipe 1 for encrypted communication.
   The application grants any request for a Host ID, thus granting pairing.
   Unencrypted packets can be received on pipe 2.
   When the Host has received data, the content of the first payload byte is output to the LEDs.

The Host automatically accepts the pairing request and receives both types of packets from the Device and displays its buttons state on the LEDs.

> You can use this sample to learn how the :ref:`gzp` subsystem can be used to establish a secure link between two devices using the Gazell protocol.”

Reviewer: ```suggestion
You can use this sample to learn how to use the :ref:`gzp` subsystem to establish a secure link between two devices over the Gazell protocol.”
```

.. gzp_dynamic_pairing_overview_end

.. gzp_dynamic_pairing_ui_start

User interface
**************

Device
======

Button 1-4:
   The button pressed state bitmask is sent to the other kit.
   A button pressed is sent as 0 and a button released is sent as 1.

Host
====

LED 1-4:
   Indicate that packets are received.
   A LED is turned off when the corresponding button is pressed on the other kit.

.. gzp_dynamic_pairing_ui_end

.. gzp_dynamic_pairing_building_start

Building and running
********************

The Device sample is under :file:`samples/gazell/gzp_dynamic_pairing/device` in the |NCS| folder structure.
The Host sample is under :file:`samples/gazell/gzp_dynamic_pairing/host` in the |NCS| folder structure.

See :ref:`gs_programming` for information about how to build and program the application.

Testing
=======

> After programming the Device sample on one of the development kits and the Host sample on the other kit, test them by performing the following steps:

Reviewer: ```suggestion
After programming the Device sample on one of the development kits and the Host sample on the other kit, complete the following steps to test them:
```

1. Power on both kits.
#. Observe that all the LEDs are off on both kits.
#. Place the kits next to each other for Gazell pairing.
#. Observe that the Host sample turns on all LEDs.

>    It indicates that the pairing is done.

Reviewer: Same comment as earlier about additional info to the step.

.. _gzp_dynamic_pairing_host:

Gazell Dynamic Pairing Host
###########################

This sample demonstrates the functionality of the :ref:`gzp` subsystem.

> It consists of two applications, one running on the Device and one running on the Host.

Reviewer 2: As above, maybe this line should say it is a single application now.
